File: front/src/models/filterModels/matchFilterModel.ts
import {Hero} from "../Hero";

export class MatchFilterModel {
    minDurationInMinutes?: number | undefined = 10;
    maxDurationInMinutes?: number | undefined = 80;
    minStartedMillisecondsBefore?: number | undefined;
    maxStartedMillisecondsBefore?: number | undefined;
    
    selfTeam: Hero[] = [];
    otherTeam: Hero[] = [];

    skip?: number | undefined;
    take?: number | undefined;
    daysAgo?: number | undefined;
}

File: Dota.API/Presentation/Match/DTO/MatchFilterModel.cs
using Newtonsoft.Json;

namespace Dota.API.Models.FilterModels;

public class MatchFilterModel
{
    [JsonProperty("minDurationInMinutes")]
    public int? MinDurationInMinutes { get; set; }

    [JsonProperty("maxDurationInMinutes")]
    public int? MaxDurationInMinutes { get; set; }

    [JsonProperty("minStartedMillisecondsBefore")]
    public long? MinStartedMillisecondsBefore { get; set; }

    [JsonProperty("maxStartedMillisecondsBefore")]
    public long? MaxStartedMillisecondsBefore { get; set; }

    [JsonProperty("selfTeam")]
    public List<Domain.Mongo.API.Hero> SelfTeam { get; set; } = [];

    [JsonProperty("otherTeam")]
    public List<Domain.Mongo.API.Hero> OtherTeam { get; set; } = [];

    [JsonProperty("skip")]
    public int? Skip { get; set; }

    [JsonProperty("take")]
    public int? Take { get; set; }

    [JsonProperty("daysAgo")]
    public int? DaysAgo { get; set; }
}